### PR TITLE
Added OptionalPassthrough transformer

### PR DESCRIPTION
**Pull request by aiwalter**
#### Reference Issues/PRs
#747 

#### What does this implement/fix? Explain your changes.
Enable tuning of implicite hyperparameters of transformers, see #747
##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [x] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.

##### For new estimators
- [x] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

___
Martin Walter <martin_friedrich.walter@daimler.com>, Mercedes-Benz AG on behalf of Daimler TSS GmbH.
https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md

### CODE REVIEW

**Comment by aiwalter** [comment]
@mloning @fkiraly
this is ready for review :)
___
Martin Walter <martin_friedrich.walter@daimler.com>, Mercedes-Benz AG on behalf of Daimler TSS GmbH.
https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md

File: sktime/transformations/series/compose.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+from sktime.transformations.base import _SeriesToSeriesTransformer
+from sktime.utils.validation.series import check_series
+
+from sklearn.base import clone
+from sklearn.utils.metaestimators import if_delegate_has_method
+
+
+class OptionalPassthrough(_SeriesToSeriesTransformer):
+    """A transformer to tune the implicite hyperparameter whether to use a
```
Comment (mloning) [inline sktime/transformations/series/compose.py:10]:
```suggestion
    """A transformer to tune the implicit hyperparameter whether or not to use a
```

File: sktime/transformations/series/compose.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    particular transformer inside a pipeline (e.g. TranformedTargetForecaster)
+    or not.This is achived by having the additional hyperparameter
+    \"passthrough\" which can be added to a grid then (see example).
```
Comment (mloning) [inline sktime/transformations/series/compose.py:13]: single quotation marks should work?

File: sktime/transformations/series/compose.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    Examples
+    ----------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> from sktime.transformations.series.compose import OptionalPassthrough
+    >>> from sktime.transformations.series.boxcox import BoxCoxTransformer
+    >>> from sktime.transformations.series.detrend import Deseasonalizer
+    >>> from sktime.forecasting.compose import TransformedTargetForecaster
+    >>> from sktime.forecasting.model_selection import (
+        ForecastingGridSearchCV,
```
Comment (mloning) [inline sktime/transformations/series/compose.py:32]: I think we have to add `...` for continued lines

File: sktime/transformations/series/compose.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+    def __init__(self, transformer, passthrough=True):
+        self.transformer = transformer
+        self.transformer_ = None
+        self.passthrough = passthrough
+        self._is_fitted = False
+        super(OptionalPassthrough, self).__init__()
+
+    def fit(self, Z, X=None):
+        self.transformer_ = clone(self.transformer)
```
Comment (mloning) [inline sktime/transformations/series/compose.py:75]: I guess this can be moved into the if clause

File: sktime/transformations/series/compose.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        self.check_is_fitted()
+        z = check_series(Z, enforce_univariate=True)
+        if not self.passthrough:
+            z = self.transformer_.transform(z, X)
+        return z
+
+    @if_delegate_has_method(delegate="transformer")
+    def inverse_transform(self, Z, X=None):
+        self.check_is_fitted()
+        z = check_series(Z, enforce_univariate=True)
```
Comment (mloning) [inline sktime/transformations/series/compose.py:91]: make it `Z` or change below (capital letter)

**Review by mloning — COMMENTED** [review]
Let's wait for #766 to be merged first to make sure the examples in the docstrings pass in this PR.

File: sktime/transformations/series/compose.py (hunk 3 above)
Comment (aiwalter) [reply sktime/transformations/series/compose.py:32]: @mloning  where exactly? I used a docstring example also [here ](https://www.sktime.org/en/latest/api_reference/modules/auto_generated/sktime.forecasting.model_evaluation.evaluate.html) without `...` but seems the indentation is not perfectly. I couldnt find any reference of how to exactly write such example code snippets with Numpy docstring style, do you have any reference here?

Comment (aiwalter) [reply sktime/transformations/series/compose.py:32]: maybe we better remove `>>> ` from it to make the code copy-pasteable on the website?

Comment (mloning) [reply sktime/transformations/series/compose.py:38]: The `>>>` is necessary, I also couldn't find any style guide, so trail and error, it's very similar to the IPython command line, Pycharm also seemed to highlight bad code snippets.

Comment (mloning) [reply sktime/transformations/series/compose.py:38]: I think you add this to the website to toggle them on and off - this is what scikit-learn does: check out what happens when you hover over the examples section: https://scikit-learn.org/stable/modules/generated/sklearn.dummy.DummyClassifier.html

**Comment by aiwalter** [comment]
@mloning great feature! We should ideally have a lot more such docstring examples, maybe some Mentees can work on it? 😄 . I will also try to add more of them to stuff I have contributed.
___
Martin Walter <martin_friedrich.walter@daimler.com>, Mercedes-Benz AG on behalf of Daimler TSS GmbH.
https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md

**Review by mloning — APPROVED** [review]

**Pull request merged**